src/game/game.c

```diff
@@ -25,14 +25,26 @@ bool StartGame(int32_t level_num, GAMEFLOW_LEVEL_TYPE level_type)
 {
     g_CurrentLevel = level_num;
     g_GameInfo.current_level_type = level_type;
-    if (level_type == GFL_SAVED) {
+    if (level_type == GFL_SAVED || level_type == GFL_RESTART) {
         // reset start info to the defaults so that we do not do
         // GlobalItemReplace in the inventory initialization routines too early
         Savegame_ResetStartInfo(level_num);
     } else {
         InitialiseLevelFlags();
```

Comment:
Do `Savegame_ResetStartInfo` and `InitialiseLevelFlags` need to be revisited now that we have start and current info for every level?